src/main/java/zd/zero/waifu/motivator/plugin/integrations/ErrorReporter.kt

```diff
@@ -0,0 +1,137 @@
+package zd.zero.waifu.motivator.plugin.integrations
+
+import com.google.gson.GsonBuilder
+import com.intellij.ide.IdeBundle
+import com.intellij.ide.plugins.PluginManagerCore
+import com.intellij.ide.ui.LafManager
+import com.intellij.openapi.application.ApplicationNamesInfo
+import com.intellij.openapi.application.ex.ApplicationInfoEx
+import com.intellij.openapi.application.impl.ApplicationInfoImpl
+import com.intellij.openapi.diagnostic.ErrorReportSubmitter
+import com.intellij.openapi.diagnostic.IdeaLoggingEvent
+import com.intellij.openapi.diagnostic.SubmittedReportInfo
+import com.intellij.openapi.util.SystemInfo
+import com.intellij.openapi.util.io.FileUtilRt
+import com.intellij.openapi.util.registry.Registry
+import com.intellij.util.Consumer
+import com.intellij.util.text.DateFormatUtil
+import io.sentry.DefaultSentryClientFactory
+import io.sentry.SentryClient
+import io.sentry.dsn.Dsn
+import io.sentry.event.Event
+import io.sentry.event.EventBuilder
+import io.sentry.event.UserBuilder
+import zd.zero.waifu.motivator.plugin.settings.WaifuMotivatorPluginState
+import zd.zero.waifu.motivator.plugin.tools.RestClient
+import java.awt.Component
+import java.lang.management.ManagementFactory
+import java.text.SimpleDateFormat
+import java.util.*
+import java.util.stream.Collectors
+
+class ErrorReporter : ErrorReportSubmitter() {
+    override fun getReportActionText(): String = "Report Anonymously"
+
+    companion object {
+        private val gson = GsonBuilder().setPrettyPrinting().create()
+        private val sentryClient: SentryClient =
+            DefaultSentryClientFactory().createSentryClient(Dsn(
+                RestClient.performGet(
+                    "https://jetbrains.assets.unthrottled.io/waifu-motivator/sentry-dsn.txt"
```

Comment:
I have this set up. That way if something goes rouge, I can always change the configuration to point to a different project.